spec/lib/assert_matchers_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "spec_helper"
+require "govuk_schemas/assert_matchers"
+
+RSpec.describe GovukSchemas::AssertMatchers do
+  include GovukSchemas::AssertMatchers
+
+  def assert(boolean, error_message); end
+
+  %w[publisher links frontend notification].each do |schema_type|
+    describe "#assert_valid_against_#{schema_type}_schema" do
+      it "detects an valid schema" do
+        example = GovukSchemas::RandomExample.for_schema("#{schema_type}_schema": "placeholder")
+        validator = GovukSchemas::Validator.new("placeholder", schema_type, example)
+        expect(self).to receive(:assert).with(true, validator.error_message)
+
+        send("assert_valid_against_#{schema_type}_schema", example, "placeholder")
```

Comment:
nitpick: public_send is nicer to use than send as it shows an intent not to use private methods: https://github.com/rubocop/ruby-style-guide#prefer-public-send

Not sure why it's not in our styleguide 🤔

Reply:
I also didn't know that #send broke class security! 🤯